### ISSUE TITLE
Potential Gradient Error when Reloading Frozen Weights in `qmodule.py` `_load_from_state_dict`

### ISSUE THREAD

**Issue by cjfghk5697** [huggingface/optimum-quanto#293]
There is a potential issue in the `_load_from_state_dict` method where reloading frozen weights into a frozen module might cause a gradient-related error. The FIXME comment in the code points out this problem.

```python
if type(self.weight.data) is not type(deserialized_weight):
    # Reloading frozen weights into unfrozen module: move to the correct device and force assignment
    self.weight = torch.nn.Parameter(deserialized_weight.to(self.weight.device))
else:
    # FIXME: here we should copy frozen weights into frozen module, but this leads to grad error
    self.weight = torch.nn.Parameter(deserialized_weight.to(self.weight.device))
```

Proposed Solution:

Explicitly set the requires_grad state of the reloaded parameters to False to prevent errors.
Alternatively, use the .data attribute to directly copy the values without altering the requires_grad state.

```python
self.weight = torch.nn.Parameter(deserialized_weight.to(self.weight.device))
self.weight.requires_grad = False
```

or

```python
self.weight.data.copy_(deserialized_weight.to(self.weight.device).data)
```

This fix would prevent potential gradient errors when reloading frozen weights into a frozen module, ensuring the operation is safe and consistent.
